### PR TITLE
ROX-21127: Refactor cluster init bundles pages

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

Prerequisite to add cluster init bundle wizard, although discussion continues about its contents.

By the way, there is only an endpoint to fetch bundles, not to fetch a bundle by id:
https://github.com/stackrox/stackrox/blob/master/proto/api/v1/cluster_init_service.proto#L82-L86

### Residue

1. InitBundlePage: Investigate second argument of service function and add confirmation modal for **Revoke bundle**.
2. InitBundlesRoute: Replace replace resources with Admin role.
3. Write integration tests.

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

## Testing Performed

In stackrox/ui folder before `yarn deploy`

```sh
export ROX_MOVE_INIT_BUNDLES_UI=true
```

### Manual testing

1. Visit /main/clusters/init-bundles
    clusters_init-bundles

2. Click link to /main/clusters/init-bundles/id
    clusters_init-bundles_id

### CODE REVIEW

File: ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlesHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+} from '@patternfly/react-core';
+
+import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
+import PageTitle from 'Components/PageTitle';
+import { clustersBasePath, clustersInitBundlesPath } from 'routePaths';
+
+const titleInitBundles = 'Cluster init bundles';
+
+export type InitBundlesHeaderProps = {
+    alignRightElement?: ReactElement | null;
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlesHeader.tsx:19]: If there is the potential for multiple buttons in the future, what do you think about `headerActions?: ReactElement[] | null;` as an alternative naming? That would abstract over the positioning and focus more on child component intent.

File: ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlesPage.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                        variant="warning"
+                        title="Unable to fetch cluster init bundles"
+                        component="div"
+                        isInline
+                    >
+                        {getAxiosErrorMessage(errorForFetch)}
+                    </Alert>
+                ) : (
+                    <InitBundlesTable initBundles={dataForFetch?.response?.items ?? []} />
+                )}
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlesPage.tsx:54]: IMO this is probably the cleanest use of nested ternaries, and probably more clear than assigning the rendered content to an intermediate variable with an `if/else` chain.

**Review by dvail — APPROVED** [review]
A few comments for consideration regarding naming, but otherwise LGTM!

File: ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlePage.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                Revoke bundle
+            </Button>
+        ) : null;
+
+    /* eslint-disable no-nested-ternary */
+    return (
+        <>
+            <InitBundlesHeader
+                alignRightElement={alignRightElement}
+                titleNotInitBundles="Cluster init bundle"
```
[the remark this answers is not in the document]
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlePage.tsx:62]: Aha, this page has singular **bundle** compared to plural **bundles** for the default title.

File: ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlesHeader.tsx (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlesHeader.tsx:19]: Yes, indeed. What do you think about `ReactNode` versus `ReactElement[]` for arbitrary content?

File: ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlesRoute.tsx [hunk 4]
```diff
@@ -19,79 +15,26 @@ function hasCreateAction(search: string) {
 
 function InitBundlesRoute(): ReactElement {
     const { hasReadWriteAccess } = usePermissions();
-    // Pending resolution whether resources or Admin role.
+    // TODO replace resources with Admin role.
```
[the remark this answers is not in the document]
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Clusters/InitBundles/InitBundlesRoute.tsx:18]: Yes, Alex just confirmed that the requests require **Admin** role.

**Comment by pedrottimark** [comment]
gke-ui-e2e-tests failure

> Cannot get catalogsources in kube-system: error: the server doesn't have a resource type "catalogsources"
> Cannot get subscriptions in kube-system: error: the server doesn't have a resource type "subscriptions"
> Cannot get clusterserviceversions in kube-system: error: the server doesn't have a resource type "clusterserviceversions"

**Pull request merged**